search/views.py

```diff
@@ -37,6 +38,224 @@ def error404(request):
+                        try:
+                            val = value.pop()
+                            value.push(calculate(val,-1,opr))
+                        except:
+                            return False
+                    elif opr in op_precdn:
+                        try:
+                            val2 = value.pop()
+                            val1 = value.pop()
+                            value.push(calculate(val1,val2,opr))
+                        except:
+                            return False
+                operator.pop()
+                open_bracket -= 1
+            else:
+                return False
+        elif i not in op_trigno and i not in op_precdn:
+            value.push(i)
+        else:
+            if operator.isEmpty():
+                operator.push(i)
+            else:
+                opr = operator.peek()
+                if opr =='(':
+                    operator.push(i)
+                elif opr in op_trigno or opr=='u+' or opr=='u-':
+                    if not value.isEmpty():
+                        try:
+                            val = value.pop()
+                            operator.pop()
+                            operator.push(i)
+                            value.push(calculate(val,-1,opr))
+                        except:
+                            return False
+                elif opr in op_precdn:
+                    if value.size()>1:
+                        if check_precedence(opr,i,op_precdn,op_trigno):
+                            try:
+                                val2 = value.pop()
+                                val1 = value.pop()
+                                operator.pop()
+                                value.push(calculate(val1,val2,opr))
+                            except:
+                                return False
+                            operator.push(i)
+                        else:
+                            operator.push(i)
+                    else:
+                        operator.push(i)
+                else:
+                    return False
+
+    while value.size()!=0 and operator.size()!=0:
+        opr = operator.pop()
+        if opr in op_trigno or opr=='u+' or opr=='u-':
+            val = value.pop()
+            try:
+                value.push(calculate(val,-1,opr))
+            except:
+                return False
+        else:
+            if value.size()>1:
+                try:
+                    val2 = value.pop()
+                    val1 = value.pop()
+                    value.push(calculate(val1,val2,opr))
+                except:
+                    return False
+            else:
+                break
+    if value.size()==1 and operator.size()==0:
+        return value.peek()
+    else:
+        return False
+
+def getResult(expr):
+    op_precdn = ['u+','u-','^', '/', '*', '%', '+', '-']
+    op_trigno = ['sin', 'cos', 'cosec', 'tan', 'sec', 'cot', 'exp', 'sqrt']
```

Comment:
Calling this trigno is a bit misleading (considering trigno). Maybe just call it functions? Also, you might consider adding an option for specifying degrees/radians (using deg/rad as a unit for these trig functions)